include/ginkgo/core/base/math.hpp

```diff
@@ -133,6 +133,27 @@ struct remove_complex_impl<std::complex<T>> {
 };
 
 
+/**
+ * Use the complex type if it is not complex.
+ *
+ * @tparam T  the type being made complex
+ */
+template <typename T>
+struct make_complex_impl {
```

Comment:
Maybe rename this to `add_complex` since this would be the more natural antonym ("opposite") to `remove_complex`